src/main/java/it/gov/pagopa/payment/service/payment/common/CommonPreAuthServiceImpl.java

```diff
@@ -62,13 +67,11 @@ public AuthPaymentDTO previewPayment(TransactionInProgress trx, String channel)
               trx.getId(), trx.getUserId(), preview.getRejectionReasons(), channel);
       log.info("[TRX_STATUS][REJECTED] The transaction with trxId {} trxCode {}, has been rejected ",trx.getId(), trx.getTrxCode());
       if (preview.getRejectionReasons().contains(RewardConstants.INITIATIVE_REJECTION_REASON_BUDGET_EXHAUSTED)) {
-        throw new ClientExceptionWithBody(
-                HttpStatus.FORBIDDEN,
+        throw new BudgetExhaustedException(
```

Comment:
some Exception, like this, could have a default code that is supposed to be always the same: we could thus expose a constructor without CODE, using such default code in the constructor